database.cc

```diff
@@ -2321,6 +2321,71 @@ future<> database::drain() {
+            make_lw_shared<lister::dir_entry_types>(lister::dir_entry_types{directory_entry_type::directory});
+        std::unordered_map<sstring, std::unordered_map<std::pair<sstring, sstring>, snapshot_ctl::snapshot_details, utils::tuple_hash>> details;
+
+        for (auto& datadir : data_dirs) {
+            lister::scan_dir(datadir, *dirs_only_entries_ptr, [this, dirs_only_entries_ptr, &details] (fs::path parent_dir, directory_entry de) mutable {
+                // KS directory
+                sstring ks_name = de.name;
+
+                return lister::scan_dir(parent_dir / de.name, *dirs_only_entries_ptr, [this, dirs_only_entries_ptr, &details, ks_name = std::move(ks_name)] (fs::path parent_dir, directory_entry de) mutable {
+                    // CF directory
+                    sstring cf_name = de.name.substr(0, de.name.find("-"));
+                    fs::path cf_dir = parent_dir / de.name;
+
+                    return lister::scan_dir(parent_dir / de.name, *dirs_only_entries_ptr, [this, dirs_only_entries_ptr, &details, ks_name, cf_name = std::move(cf_name), cf_dir = std::move(cf_dir)] (fs::path parent_dir, directory_entry de) mutable {
+                        // "snapshots" directory
+
+                        return lister::scan_dir(parent_dir / de.name, *dirs_only_entries_ptr, [this, &details, ks_name, cf_name, cf_dir] (fs::path parent_dir, directory_entry de) {
+                            sstring snapshot_name = de.name;
+                            std::pair<sstring, sstring> cf_map_key = { ks_name, cf_name };
+                            std::unordered_map<std::pair<sstring, sstring>, snapshot_ctl::snapshot_details, utils::tuple_hash> cf_map = {{ cf_map_key, {0, 0, cf_name, ks_name} }};
+                            details.emplace(snapshot_name, std::move(cf_map));
+
+                            return lister::scan_dir(parent_dir / de.name,  { directory_entry_type::regular }, [this, &details, ks_name, cf_name, cf_map_key, cf_dir, snapshot_name = std::move(snapshot_name)] (fs::path snapshot_dir, directory_entry de) {
+                                return io_check(file_size, (snapshot_dir / de.name).native()).then([this, cf_dir, cf_map_key, &details, snapshot_name, snapshot_dir, name = de.name] (auto size) {
+                                    // The manifest is the only file expected to be in this directory not belonging to the SSTable.
+                                    // For it, we account the total size, but zero it for the true size calculation.
+                                    //
+                                    // All the others should just generate an exception: there is something wrong, so don't blindly
+                                    // add it to the size.
+                                    if (name != "manifest.json" && name != "schema.cql") {
+                                        sstables::entry_descriptor::make_descriptor(snapshot_dir.native(), name);
```

Comment:
Why is this called when its returned type is ignored?

Reply:
looks like a leftover from a previous version, removed it

Reply:
I just want to make sure that it's not validating something and throwing exception when something's wrong. I can see that it was called like this in the past as well https://github.com/scylladb/scylla/blob/3226c5bf9d6db6c3f3c57008aacab9cab8c57fc9/table.cc#L1473 so I'm not sure if it's an old code that should be removed or something important. @bhalevy do you know?

Reply:
All other usages of the function do something with the return type.